src/main/java/UseCase/NextSong.java

```diff
@@ -0,0 +1,39 @@
+package UseCase;
+
+import Entities.MusicPlayer;
+import Entities.Song;
+import InputBoundary.NextSongInputBoundary;
+import InputBoundary.playPlaylistInputBoundary;
+import InputData.playlistInputData;
+import OutputBoundary.songOutputBoundary;
+
+import java.util.ArrayList;
+
+public class NextSong implements NextSongInputBoundary {
+    private final ArrayList<Song> playlist;
+    private final String name;
+    private final songOutputBoundary presenter;
+
+    public NextSong(playlistInputData data, songOutputBoundary presenter){
+        this.playlist = data.getSongs();
+        this.presenter = presenter;
+        this.name = data.getName();
+    }
+
+    @Override
+    public void skipSong() {
+        MusicPlayer mp = MusicPlayer.getInstance();
+        int id;
+        if (mp.getCurrentSong() != null)
+            id = playlist.indexOf(mp.getCurrentSong());
```

Comment:
if you call the presenter in the else branch, why aren't you calling the presenter here? if it's handled in play.play(), could you add a comment saying so?

Reply:
fixed